src/components/form/field-controls/checkbox-field/index.tsx

```diff
@@ -13,11 +12,10 @@ const CheckboxField = ({
 	label,
 	labelFontWeight,
 	name,
-	onChange = () => null, // Default value makes this easier to invoke w/o extra logic
-	onClick,
+	onClick = () => void 0,
+	onChange = () => void 0,
 }: CheckboxFieldProps) => {
 	const inputId = useId(id)
-	const [isChecked, setIsChecked] = useState<boolean>(checked)
```

Comment:
I removed the internal state here to leave it as a completely controlled component. This makes it easier to deal with in practice and removes the need to sync the `checked` prop with the internal state value